### PR TITLE
[SPARK-43080][BUILD] Upgrade `zstd-jni` to 1.5.5-1

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr aims upgrade `zstd-jni` from 1.5.4-2 to 1.5.5-1.

### Why are the changes needed?
New version includes a bug fix about `Closing ZstdOutputStream without any write produces empty files`:
- https://github.com/luben/zstd-jni/issues/249  | https://github.com/luben/zstd-jni/commit/2b6f2eca3e07817f4713ec43a047100c26cc51a1

Other changes as follows:
- https://github.com/luben/zstd-jni/compare/v1.5.4-2...v1.5.5-1

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### CODE REVIEW

**Comment by LuciferYang** [comment]
Please wait for me to check and update the benchmark results of `ZStandardBenchmark`

File: core/benchmarks/ZStandardBenchmark-results.txt [hunk 1]
```diff
@@ -2,26 +2,26 @@
...
-Compression 10000 times at level 3 without buffer pool            639            640           1          0.0       63949.2       0.6X
-Compression 10000 times at level 1 with buffer pool               192            197           3          0.1       19199.2       2.1X
-Compression 10000 times at level 2 with buffer pool               243            250           4          0.0       24310.4       1.6X
-Compression 10000 times at level 3 with buffer pool               417            431           9          0.0       41711.5       0.9X
+Compression 10000 times at level 1 without buffer pool            277            348         107          0.0       27655.0       1.0X
+Compression 10000 times at level 2 without buffer pool            320            321           1          0.0       32020.3       0.9X
+Compression 10000 times at level 3 without buffer pool            460            464           3          0.0       46009.5       0.6X
+Compression 10000 times at level 1 with buffer pool               257            258           1          0.0       25654.4       1.1X
```
Comment (dongjoon-hyun) [inline core/benchmarks/ZStandardBenchmark-results.txt:12]: Is this from `2.1x` to `1.1x`?

**Review by dongjoon-hyun — COMMENTED** [review]
New results look reasonable.

**Comment by LuciferYang** [comment]
The CPU model used in micro-bench is the same as before, and the results have not changed much now

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @LuciferYang .
Merged to master.

**Pull request closed without merging**